model.py

```diff
@@ -3270,10 +3245,9 @@ def language_code(self):
 
     def all_cover_images(self):
         _db = Session.object_session(self)
-        primary_identifier_ids = [
-            x.primary_identifier.id for x in self.editions]
+        primary_identifier_id = self.presentation_edition.primary_identifier.id
```

Comment:
Same deal here, and below. Maybe create an internal method that gets the starter set of identifier IDs.